core/src/main/java/org/mule/runtime/core/api/processor/factory/AsynchronousProcessingStrategyFactory.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.processor.factory;
```

Comment:
Don't think this is the best package name as these are not processor factories.  I'd put these in the existing `.strategy.` package or maybe `.strategy.factory`, but I don't think that's necessary really.